src/internal/grpcutil/iter.go

```diff
@@ -0,0 +1,37 @@
+package grpcutil
+
+import (
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"google.golang.org/grpc"
+)
+
+type ClientStream[T any] interface {
+	Recv() (*T, error)
```

Comment:
Can this be T instead of *T (then the pointer nature is encoded by the type constraint above, and iterator[t] will return slices of t instead of *t)?

Reply:
Does that work with `RecvMsg`?  I think we would get a double pointer.

Would it be better if`grpcutil.Iterator[T]` implements `stream.Iterator[*T]`?

Reply:
So I guess my thought is that the type variable should directly match the return type of RecvMsg; right now we are assuming that exists, so the person creating an iterator has to know that they need to use a different type than what's actually in the API if that makes sense.  i.e if RecvMsg returns (Foo, error), they can't use the iterator.  It never does, of course.

BTW, I wonder if `proto.Message` is a better type constraint than `any`.

Reply:
I agree with `proto.Message` vs `any`, but I think that means `grpcutil.Iterator[T]` implements `stream.Iterator[*T]`, which means that it's Next method looks like this `Next(ctx, **T)`.  I'm not really sure if that's a good idea yet; maybe it should be private for now.

I'm more confident about the following:
- `stream.Iterator` taking a pointer to T for Next is the right API.
- functions in grpcutil should work with pointers to protocol buffer objects because that's how all the other protocol buffer APIs work.   e.g. `Collect[T](cs ClientStream, max int) ([]*T, error)`